### PR TITLE
e2e/virtual/syncer: fixing race with resource controller in TestSyncerVirtualWorkspace

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/kcp-dev/kcp/issues/1187.

### CODE REVIEW

File: test/e2e/virtual/syncer/virtualworkspace_test.go [hunk 1]
```diff
@@ -380,14 +381,15 @@ func TestSyncerVirtualWorkspace(t *testing.T) {
...
-				for _, cs := range virtualWorkspaceCowboys.Items {
-					if logicalcluster.From(&cs) == wildwestClusterName {
-						require.Equal(t, "luckyluke", cs.Name)
-					}
-				}
+				t.Log("Wait for resource controller to schedule cowboy and then show up via virtual workspace wildcard request")
+				var cowboys *wildwestv1alpha1.CowboyList
+				require.Eventually(t, func() bool {
+					cowboys, err = vwClusterClient.Cluster(logicalcluster.Wildcard).WildwestV1alpha1().Cowboys("").List(ctx, metav1.ListOptions{})
+					require.NoError(t, err)
```
Comment (ncdc) [inline test/e2e/virtual/syncer/virtualworkspace_test.go:388]: Can you do a require inside an Eventually without it failing?

Comment (sttts) [reply test/e2e/virtual/syncer/virtualworkspace_test.go:388]: we have that everywhere. If it does not work, we should generally fix it.

Comment (ncdc) [reply test/e2e/virtual/syncer/virtualworkspace_test.go:388]: Compare
```go
func TestAndy(t *testing.T) {
	i := 0

	require.Eventually(t, func() bool {
		i++
		require.True(t, i%2 == 0)
		return true
	}, time.Second, 100*time.Millisecond)
}
```

with

```go
func TestAndy(t *testing.T) {
	i := 0

	require.Eventually(t, func() bool {
		i++
		return i%2 == 0
	}, time.Second, 100*time.Millisecond)
}
```

Both should eventually succeed, right? But only the 2nd one does.

Comment (sttts) [reply test/e2e/virtual/syncer/virtualworkspace_test.go:388]: Why should the first succeed? It should not.

**Pull request merged**